test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -37,6 +38,35 @@ func StatefulSetIsReady(mdb *mdbv1.MongoDB) func(t *testing.T) {
 	}
 }
 
+func StatefulSetHasOwnerReference(mdb *mdbv1.MongoDB) func(t *testing.T) {
+	return func(t *testing.T) {
+		sts := appsv1.StatefulSet{}
+		err := f.Global.Client.Get(context.TODO(), types.NamespacedName{Name: mdb.Name, Namespace: f.Global.OperatorNamespace}, &sts)
+		if err != nil {
+			t.Fatal(fmt.Errorf("%s", err))
+		}
+
+		ownerReferenceMdb :=
```

Comment:
one thing that to consider is that we have no way currently of testing for a different owner reference (we do only have one so it's not that big a deal) but a possible option here is to add it as a function argument so in the test we can decide which owner reference to look for e.g.

```golang
StatefulSetHasOwnerReference(mdb *mdbv1.MongoDB, expectedOwnerReference metav1.OwnerReference)
```

will leave this one up to you!